bundles/tools.vitruv.change.propagation/src/tools/vitruv/change/propagation/impl/DefaultChangeRecordingModelRepository.java

```diff
@@ -0,0 +1,190 @@
+package tools.vitruv.change.propagation.impl;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+
+import tools.vitruv.change.composite.description.TransactionalChange;
+import tools.vitruv.change.composite.description.VitruviusChange;
+import tools.vitruv.change.composite.recording.ChangeRecorder;
+import tools.vitruv.change.correspondence.CorrespondenceModel;
+import tools.vitruv.change.correspondence.InternalCorrespondenceModel;
+import tools.vitruv.change.propagation.PersistableChangeRecordingModelRepository;
+
+import static edu.kit.ipd.sdq.commons.util.org.eclipse.emf.ecore.resource.ResourceSetUtil.withGlobalFactories;
+import static edu.kit.ipd.sdq.commons.util.org.eclipse.emf.ecore.resource.ResourceSetUtil.getOrCreateResource;
+import static edu.kit.ipd.sdq.commons.util.org.eclipse.emf.ecore.resource.ResourceSetUtil.loadOrCreateResource;
+import static tools.vitruv.change.correspondence.CorrespondenceModelFactory.createCorrespondenceModel;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+
+/**
+ * A default implementation of a {@link PersistableChangeRecordingModelRepository}. It manages a {@link ResourceSet} for
+ * the models, which are automatically monitored for changes. It loads existing models calling
+ * {@link #loadExistingModels()} by loading the correspondence model at the URI specified in the constructor and
+ * resolving all models references in the correspondence model. If no such URI is specified, loading will have no
+ * effect.
+ */
+public class DefaultChangeRecordingModelRepository implements PersistableChangeRecordingModelRepository {
+	private static final Logger LOGGER = Logger.getLogger(DefaultChangeRecordingModelRepository.class);
+	private final ResourceSet modelsResourceSet;
+	private final InternalCorrespondenceModel correspondenceModel;
+	private final ChangeRecorder changeRecorder;
+	private final Path consistencyMetadataFolder;
+
+	private boolean isLoading = false;
+
+	/**
+	 * Creates a repository without a defined persistence for the correspondence model and using a temporary folder for
+	 * storing consistency metadata.
+	 * @throws IOException when no temporary metadata folder can be instantiated
+	 */
+	public DefaultChangeRecordingModelRepository() throws IOException {
```

Comment:
Is this constructor only for convenient testing? It does not look like you can use it productively?

Reply:
True, it is only usable for test purposes. I will move it to an according project.

Reply:
I have fixed that in 140b34e.